### DISCUSSION TITLE
Run GitHub Actions on forked repository for pushing certain changes

### DISCUSSION THREAD

**Discussion by uditgaurav** [community/community#26865] [Actions]
<p>Hi everyone,</p>
<p>This is more of a question for everyone. I want to run GitHub Action on the forked repo for a particular PR. The action should push something on the forked repo before merging the PR. Is this possible?<br>
I know that the base repo will not have the permission to push in the forked repo so we need to run the actions of forked repo for this use case. So how could we do that?</p>
<p>I was trying with the following yaml:</p>
<pre><code class="lang-yaml">name: Actions-CI
on:
  issue_comment:
    types: [created]

jobs:
  tests:
    if: contains(github.event.comment.html_url, '/pull/') &amp;&amp; startsWith(github.event.comment.body, '/run-e2e')
    runs-on: ubuntu-latest
    steps:

      #Using the last commit id of pull request
      - uses: octokit/request-action@v2.x
        id: get_PR_commits
        with:
          route: GET /repos/:repo/pulls/:pull_number/commits
          repo: ${{ github.repository }}
          pull_number: ${{ github.event.issue.number }}
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

      - name: set commit to output
        id: getcommit
        run: | 
           prsha=$(echo $response | jq '.[-1].sha'  | tr -d '"')
           echo "::set-output name=sha::$prsha" 
        env: 
          response:  ${{ steps.get_PR_commits.outputs.data }}
          
# checkout on the lastest commit 
      - uses: actions/checkout@v2
        with:
          ref: ${{steps.getcommit.outputs.sha}}

# setup golang
      - uses: actions/setup-go@v2
        with:
          stable: 'false'
          go-version: '1.14.0'

# run embedmd (this will write something in the file)
      - name: Installing and Running embedmed
        run: |
          go get github.com/campoy/embedmd
          cd docs
          embedmd -w $(find *.md)

      - name: Commit files
        run: |
          git config --local user.email "githubactions@github.com"
          git config --local user.name "Myname"
          git commit -s -m "Run embedmed" -a
      - name: Push changes
        uses: ad-m/github-push-action@master
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}

</code></pre>
<p>Thanks !</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/uditgaurav">@uditgaurav</a>,</p>
<p>When you create a PR from forked repo to base repo, the PR is created on the <code>BASE</code> repo not forked repo. Therefore, if you comment to PR, the ‘issue_comment’ will only be triggered on base repo(if you put it on base repo default branch), not on forked repo.</p>
<p>On forked repo, you need to enable github action on ‘actions’ tab firstly. And then you can create a workflow (eg: event push…etc) on forked repo, to commit changes before PR merge.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/d/d120e1e1c5ee38ca175987acbbef6170ac4fd04d.png" data-download-href="/uploads/short-url/tQ2eY55x960iPNHEI4QycAWVPOd.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/52772761/181100365-2f29ff45-4bbf-468c-8d3e-9af52f5484f5.png" alt="image" data-base62-sha1="tQ2eY55x960iPNHEI4QycAWVPOd" width="690" height="239" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/d/d120e1e1c5ee38ca175987acbbef6170ac4fd04d_2_690x239.png, https://aws1.discourse-cdn.com/github/optimized/2X/d/d120e1e1c5ee38ca175987acbbef6170ac4fd04d_2_1035x358.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/d/d120e1e1c5ee38ca175987acbbef6170ac4fd04d.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/d/d120e1e1c5ee38ca175987acbbef6170ac4fd04d_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1094×379 18.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>If you don’t have forked repo access token, you cannot commit to forked repo.</p>
<p>If you have forked repo token, on base repo, you can checkout forked branch(get info from Github Context), commit the changes to forked repo before PR merged.</p>
<p>Thanks</p>
